inspirehep/modules/orcid/utils.py

```diff
@@ -54,3 +60,378 @@ def get_authors_credentials(author):
+        index='records-authors',
+        doc_type='authors',
+        body=es_query,
+        size=10000
+    )['hits']['hits']
+    return records
+
+
+def get_identifiers_from_orcid_records(orcid, token):
+
+    api = current_app.extensions['inspire-orcid'].orcid_api
+
+    orcid_records = api.read_record_public(
+        orcid, 'activities', token)
+    orcid_records_dois = []
+    orcid_records_arxivs = []
+    for record in orcid_records['works']['group']:
+        if record['external-ids']['external-id'][0]['external-id-type'] == 'doi':
+            orcid_records_dois.append(
+                record['external-ids']['external-id'][0]['external-id-value'])
+        if record['external-ids']['external-id'][0]['external-id-type'] == 'arxiv':
+            orcid_records_arxivs.append(
+                record['external-ids']['external-id'][0]['external-id-value'])
+
+    return orcid_records_dois, orcid_records_arxivs
+
+
+def extract_author_profiles_from_hep_records(records):
+    possible_profiles = set()
+    for record in records:
+        try:
+            for author in record['inner_hits']['authors']['hits']['hits']:
+                possible_profiles.add(author['_source']['record']['$ref'])
+        except KeyError:
+            continue
+    return possible_profiles
+
+
+def solve_references_of_authors(refs):
+    possible_profiles = []
+    current_app.logger.info(refs)
+    for ref in refs:
+        author_profile = replace_refs({'$ref': ref}, source='es')
+        try:
+            # If the profile has an orcid it's already claimed by somebody.
+            for identifier in author_profile['ids']:
+                if 'orcid' in identifier:
+                    continue
+            possible_profiles.append(ref)
+        except KeyError:
+            continue
+    return possible_profiles
+
+
+def pick_records_from_author_profiles(profiles):
+    number_of_suggested_records = len(profiles)
+    number_of_records_per_profile = 1
+    picked_records = {}
+    picked_recids = []
+    if len(profiles) is 2:
+        number_of_suggested_records = 4
+        number_of_records_per_profile = 2
+    if len(profiles) > 10:
+        number_of_suggested_records = 10
+    for i, profile in enumerate(profiles):
+        picked_records[profile] = []
+        if i is number_of_suggested_records:
+            break
+        records = query_elasticsearch_for_literature_records_using_a_profile_reference(
+            profile)
+        for j, record in enumerate(records):
+            if j is number_of_records_per_profile:
+                break
+            if record['_source']['control_number'] not in picked_recids:
+                record['_source']['_profile'] = profile
+                picked_records[profile].append(record)
+                picked_recids.append(record['_source']['control_number'])
+    if len(picked_recids) is 1:
+        assign_profile(profile)
+    return picked_records
+
+
+def find_profiles(possible_profiles):
```

Comment:
All these functions need the parameters documented as there are all kinds of data structures traveling around, and for example here it is not clear what `possible_profiles` might be.